services/app-web/src/pages/StateSubmission/Documents/Documents.tsx

```diff
@@ -44,20 +44,48 @@ export const Documents = ({
     const hasValidFiles = fileItems.every(
         (item) => item.status === 'UPLOAD_COMPLETE'
     )
+    const hasMissingCategories =
+        /* fileItems must have some document category.  a contract-only submission
+       must have "CONTRACT_RELATED" as the document category. */
+        fileItems.length > 0 &&
+        (fileItems.some((docs) => docs.documentCategories.length === 0) ||
+            (isContractOnly &&
+                fileItems.some(
+                    (docs) =>
+                        !docs.documentCategories.includes('CONTRACT_RELATED')
+                )))
     const hasLoadingFiles =
         fileItems.some((item) => item.status === 'PENDING') ||
         fileItems.some((item) => item.status === 'SCANNING')
     const showFileUploadError = shouldValidate && !hasValidFiles
 
+    const documentsErrorMessages = () => {
```

Comment:
I think we’re missing checking “showFileUploadError” or “shouldValidate” somwehere here. I was seeing the “You must select a document category” error message as soon as I added a document to the list. I expect to only see that error message if I try and click Continue before filling everything out correctly. 

It’s being checked in generating the errorSummary below, which I think I’m fine with being separate from this list, but it might be worth while collecting all those errors before we get to the return and checking if we should display them in one place.

Reply:
Yeah, missing a shouldValidate for actually showing the error summary.  🤦 

Can you say what you mean by "this list"?  Want to be sure I'm understanding you correctly.

Reply:
we’ve got documentsErrorMessages and we’ve got errorSummary, those were the two lists I was talking about. Though errorSummary isn’t actually a list

Reply:
Gotcha, thanks.